source/common/config/version_converter.cc

```diff
@@ -53,15 +55,15 @@ void traverseMutableMessage(ProtoVisitor& visitor, Protobuf::Message& message, c
   }
 }
 
-// Reinterpret a Protobuf message as another Protobuf message by converting to
-// wire format and back. This only works for messages that can be effectively
-// duck typed this way, e.g. with a subtype relationship modulo field name.
+// Reinterpret a Protobuf message as another Protobuf message by converting to wire format and back.
+// This only works for messages that can be effectively duck typed this way, e.g. with a subtype
+// relationship modulo field name.
 void wireCast(const Protobuf::Message& src, Protobuf::Message& dst) {
-  // This should always succeed, since we should be supplying compatible
-  // messages here, but provide a RELEASE_ASSERT as this is off critical path
-  // and we want to learn if it fails.
-  RELEASE_ASSERT(dst.ParseFromString(src.SerializeAsString()),
-                 "Unable to deserialize during wireCast()");
+  // This should should generally succeed, but if there are malformed UTF-8 strings in a message,
+  // this can fail.
+  if (!dst.ParseFromString(src.SerializeAsString())) {
+    throw EnvoyException("Unable to deserialize during wireCast()");
```

Comment:
Are we sure that throwing an exception from here is safe in all places where this is called? Would it be better to just fix the fuzzer to not pass bad data of this type?

Reply:
I'm pretty sure it's safe for the following reasons:
1. When used on configuration ingestion, we're able to take exceptions and this is primarily where this is likely to happen, e.g. handling bootstrap, xDS, some extension.
2. When downgrading to earlier versions, we must have already upgraded, so if an exception happens, it must have occurred in (1) already.

Reply:
@mattklein123 thoughts on ^^?

Reply:
Sure sounds fine, though if this is really only to fix the fuzzer it seems like it would be better to fix it there, but up to you.